source/GLaDOS.cpp

```diff
@@ -54,11 +53,13 @@ void GLaDOS::dispose() {
  *
  * @return true if the controller was initialized successfully
  */
-bool GLaDOS::init(std::shared_ptr<ShipModel> ship, std::shared_ptr<MagicInternetBox> mib) {
+bool GLaDOS::init(std::shared_ptr<ShipModel> ship, std::shared_ptr<MagicInternetBox> mib,
+				  float lvlSize) {
 	bool success = true;
 	this->ship = ship;
 	this->mib = mib;
 	this->playerID = mib->getPlayerID();
+	size = lvlSize;
```

Comment:
Why does `GLaDOS` need to store the level size as well? Can't it get that information directly from the `Shipmodel`?

Reply:
Good point, I'll change that. GLaDOS had a FULL_CIRCLE const before I was just replacing it with the field, didn't think about dependencies.